writers_rollingfilewriter.go

```diff
@@ -138,7 +140,7 @@ var compressionTypes = map[rollingArchiveType]compressionType{
 			if err != nil {
 				return err
 			}
-			return createGzip(archiveName, tar)
+			return createGzip(archiveName, bytes.NewReader(tar))
```

Comment:
That generally makes sense to me, but who is called WriteHeaderFor with a FileInfo? Wouldn't we not necessarily know the FileInfo of a file we're streaming out of an existing tar? (or would we read out the header, call WriteHeaderFor, and then stream through the contents, one file at a time?)

Also-- at what point do we just switch to an existing library like https://github.com/mholt/archiver and do the extract+rearchive on the filesystem rather than trying to stream it all in-process?